astropy/units/format/generic.py

```diff
@@ -592,6 +600,41 @@ def _do_parse(cls, s, debug=False):
                 else:
                     raise ValueError(f"Syntax error parsing unit '{s}'")
 
+    @classmethod
+    def _validate_unit(
+        cls, unit: str, detailed_exception: bool = True, class_name: str | None = None
+    ) -> None:
+        if class_name is None:
+            class_name = cls.__name__.upper()
+        if unit not in cls._units:
+            if detailed_exception:
+                raise ValueError(
+                    f"Unit '{unit}' not supported by the {class_name} standard. "
+                    + did_you_mean_units(
+                        unit,
+                        cls._units,
+                        cls._deprecated_units,
+                        cls._to_decomposed_alternative,
+                    ),
+                )
+            raise ValueError()
+        if unit in cls._deprecated_units:
+            unit_deprecation_warning(
+                unit, cls._units[unit], class_name, cls._to_decomposed_alternative
+            )
+
+    @classmethod
+    def _to_decomposed_alternative(cls, unit: UnitBase) -> str:
+        from astropy.units.core import UnitScaleError
+
+        try:
+            return cls.to_string(unit)
+        except UnitScaleError:
+            scale = unit.scale
+            unit = copy(unit)
+            unit._scale = 1.0
+            return f"{cls.to_string(unit)} (with data multiplied by {scale})"
+
 
 # 2023-02-18: The statement in the docstring is no longer true, the class is not used
```

Comment:
Hmm, looks like another followup!

Reply:
Removing the deprecated class looks like a good task for a first-time contributor.